### PR TITLE
AAU, on the CLI, I see a command to create a project from a json file

### PR DESCRIPTION
**Pull request by theodu**
- [ ] I opened a merge request on `kili` on a branch with the same name than the branch of the current pull request and forced security tests.
- [ ] I opened a pull request on the Node playground if the changes are breaking.

### CODE REVIEW

File: kili/cli.py [hunk 1]
```diff
@@ -12,8 +17,108 @@ def cli():
...
+    """Commands to interact with a Kili project"""
+
+
+@project.command(name='create')
+@click.option('--api-key', type=str, envvar='KILI_API_KEY', required=True,
+              help='Your Api Key')
+@click.option('--interface', type=click.Path(exists=True), required=True,
+              help='Path pointing to your json interface file')
+@click.option('--title', type=str, required=True,
+              help='Title of your project')
```
Comment (PierreLeveau) [inline kili/cli.py:31]: Project title

File: kili/cli.py [hunk 2]
```diff
@@ -12,8 +17,108 @@ def cli():
...
+@click.option('--input-type', type=click.Choice(INPUT_TYPE), required=True,
+              help='type of data in your project')
```
Comment (PierreLeveau) [inline kili/cli.py:33]: Project input data type

Comment (PierreLeveau) [reply kili/cli.py:33]: and add examples of `INPUT_TYPE`

File: kili/cli.py [hunk 3]
```diff
@@ -12,8 +17,108 @@ def cli():
...
+@click.option('--description', type=str, default='',
+              help='Description of your project')
```
Comment (PierreLeveau) [inline kili/cli.py:35]: Project description

File: test/test_cli.py [hunk 4]
```diff
@@ -16,6 +16,21 @@ def mocked__projects(project_id, **_):
...
+    runner.invoke(create_project,
+                  ['--interface',
+                   "test/fixtures/image_interface.json",
+                   '--title',
+                   'Test project',
+                   '--description',
+                   'desciption',
+                   '--input-type',
+                   'IMAGE'])
+
```
Comment (PierreLeveau) [inline test/test_cli.py:32]: you can use https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once on the create_project method to check that the backend is called.

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kili/cli.py (hunk 2 above)
Comment (theodu) [reply kili/cli.py:33]: When using click.choice, the -help option automatically display all possible choices
Capture d’écran 2022-06-13 à 17 22 58

File: kili/cli.py [hunk 5]
```diff
@@ -22,6 +24,45 @@ def project():
     """Commands to interact with a Kili project"""
 
 
+@project.command(name='create')
+@click.option('--api-key', type=str, envvar='KILI_API_KEY', required=True,
```
Comment (PierreLeveau) [inline kili/cli.py:28]: see the other commands, where this option has been clarified

File: kili/mutations/asset/helpers.py [hunk 6]
```diff
@@ -309,24 +309,24 @@ def get_file_paths_to_upload(files: Tuple[str, ...],
...
         path for path in files_path if check_file_mime_type(path, input_type, False)]
     if exclude is not None:
-        files_path_to_upload = [
-            path for path in files_path_to_upload if path not in exclude]
-    files_path_to_upload.sort()
-    if len(files_path_to_upload) == 0:
+        file_paths_to_upload = [
+            path for path in file_paths_to_upload if path not in exclude]
+    file_paths_to_upload.sort()
+    if len(file_paths_to_upload) == 0:
```
Comment (PierreLeveau) [inline kili/mutations/asset/helpers.py:318]: move to the return line (L.329)

File: test/test_cli.py [hunk 7]
```diff
@@ -16,6 +16,22 @@ def mocked__projects(project_id, **_):
...
+    mocker.patch("kili.client.Kili.__init__", return_value=None)
+    mocked__create_project = mocker.patch("kili.client.Kili.create_project")
+    runner = CliRunner()
+    runner.invoke(create_project,
+                  ['--interface',
+                   "test/fixtures/image_interface.json",
+                   '--title',
+                   'Test project',
+                   '--description',
+                   'desciption',
```
Comment (PierreLeveau) [inline test/test_cli.py:29]: description

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kili/cli.py [hunk 8]
```diff
@@ -66,11 +68,51 @@ def list_project(api_key: str,
...
+def create_project(api_key: str,
+                   endpoint: str,
+                   input_type,
+                   interface: str,
+                   title: str,
+                   description: str):
+    """Create a Kili project.
+    Examples:
+        $ kili project create --project-id <project_id> \
+        --interface path/to/interface.json --input-type TEXT --title "Invoice annotation project"
```
Comment (crsegerie) [inline kili/cli.py:99]: add a tab here

File: kili/constants.py [hunk 9]
```diff
@@ -6,28 +6,30 @@
...
-  'Image': 'image/jpeg,image/png,image/bmp,image/gif,image/webp,image/x-icon,'\
-      'image/tiff,image/vnd.microsoft.icon,image/svg+xml,image/avif,image/apng',
-  'Pdf': 'application/pdf',
-  'Text': 'text/plain',
-  'TimeSeries': 'text/csv',
+    'Image': 'image/jpeg,image/png,image/bmp,image/gif,image/webp,image/x-icon,'
+    'image/tiff,image/vnd.microsoft.icon,image/svg+xml,image/avif,image/apng',
+    'Pdf': 'application/pdf',
+    'Text': 'text/plain',
+    'TimeSeries': 'text/csv',
```
Comment (crsegerie) [inline kili/constants.py:20]: when is scheduled black ticket?

File: kili/mutations/asset/helpers.py [hunk 10]
```diff
@@ -296,37 +296,37 @@ def get_file_paths_to_upload(files: Tuple[str, ...],
...
         a list of the paths of the files to upload, compatible with the project type.
     """
-    files_path = []
+    file_paths = []
     for item in files:
         if os.path.isfile(item):
-            files_path.append(item)
-            print(files_path)
+            file_paths.append(item)
+            print(file_paths)
```
Comment (crsegerie) [inline kili/mutations/asset/helpers.py:303]: Add message in the rint or remoove it

File: test/fixtures/image_interface.json [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+                    },
+                    "DEFECT_CLASS_4": {
+                        "name": "defect of class 4"
+                    }
+                },
+                "input": "radio"
+            }
+        }
+    }
+}
```
Comment (crsegerie) [inline test/fixtures/image_interface.json:31]: Add a newline at the end of the file, it's a convention for Unix design

File: kili/constants.py (hunk 9 above)
Comment (theodu) [reply kili/constants.py:20]: for this sprint!

File: kili/cli.py (hunk 8 above)
Comment (theodu) [reply kili/cli.py:99]: Yeah, this is taken in the documentation ticket that is next to come ;)

**Review by crsegerie — APPROVED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**